src/Tickets/Commerce/Admin/Singular_Order_Page.php

```diff
@@ -0,0 +1,420 @@
+	 */
+	public function render_order_details( $post ): void {
+		$this->template(
+			'order-details-metabox',
+			[
+				'order'       => tec_tc_get_order( $post ),
+				'single_page' => $this,
+			]
+		);
+	}
+
+	/**
+	 * Renders the order items metabox.
+	 *
+	 * @since TBD
+	 *
+	 * @param WP_Post $post The post object.
+	 *
+	 * @return void
+	 */
+	public function render_order_items( $post ): void {
+		$this->template(
+			'order-items-metabox',
+			[
+				'order'       => tec_tc_get_order( $post ),
+				'single_page' => $this,
+			]
+		);
+	}
+
+	/**
+	 * Get the gateway label for the order.
+	 *
+	 * @since TBD
+	 *
+	 * @param WP_Post|int $order The order post object or ID.
+	 *
+	 * @return string
+	 */
+	public function get_gateway_label( $order ): string {
+		if ( is_numeric( $order ) ) {
+			$order = tec_tc_get_order( $order );
+		}
+
+		if ( ! $order instanceof WP_Post ) {
+			return '';
+		}
+
+		$gateway = tribe( Manager::class )->get_gateway_by_key( $order->gateway );
+
+		if ( $gateway instanceof Free_Gateway ) {
+			return esc_html__( 'Free', 'event-tickets' );
+		}
+
+		if ( ! $gateway ) {
+			return esc_html( $order->gateway );
+		}
+
+		$order_url = $gateway->get_order_controller()->get_gateway_dashboard_url_by_order( $order );
+
+		if ( empty( $order_url ) ) {
+			return esc_html( $gateway::get_label() );
+		}
+
+		ob_start();
+		tribe_copy_to_clipboard_button( $order->gateway_order_id, true, __( 'Copy Payment\'s Gateway Transaction ID to your Clipboard', 'event-tickets' ) );
+		$copy_button = ob_get_clean();
+
+		return sprintf(
+			'%1$s%2$s%3$s%4$s<br>%5$s',
+			'<a class="tribe-dashicons" href="' . esc_url( $order_url ) . '" target="_blank" rel="noopener noreferrer">',
+			esc_html( $gateway::get_label() ),
+			'<span class="dashicons dashicons-external"></span>',
+			'</a>',
+			$copy_button
+		);
+	}
+
+	/**
+	 * Updates the order status.
+	 *
+	 * @since TBD
+	 *
+	 * @param int     $post_id The post ID.
+	 * @param WP_Post $post    The post object.
+	 *
+	 * @return void
+	 */
+	public function update_order_status( $post_id, $post ) {
+		if ( Order::POSTTYPE !== $post->post_type ) {
+			return;
+		}
+
+		if ( ! is_admin() ) {
+			return;
+		}
+
+		if ( ! current_user_can( 'edit_post', $post_id ) ) {
+			return;
+		}
+
+		$new_status = tribe_get_request_var( 'tribe-tickets-commerce-status', false );
+
+		if ( ! $new_status ) {
+			return;
+		}
+
+		$order = tec_tc_get_order( $post );
+
+		if ( ! $order instanceof WP_Post ) {
+			return;
+		}
+
+		$new_status = tribe( Status_Handler::class )->get_by_slug( $new_status );
+
+		if ( $new_status->get_wp_slug() === $order->post_status ) {
+			return;
+		}
+
+		$current_status = tribe( Status_Handler::class )->get_by_wp_slug( $order->post_status );
+
+		if ( ! $current_status->can_apply_to( $order, $new_status ) ) {
+			$this->redirect_with_message( $post_id, 1001 );
+			return;
+		}
+
+		$result = tribe( Order::class )->modify_status( $order->ID, $new_status->get_slug() );
```

Comment:
Can we please confirm that the status transition actions are being fired here? Otherwise, we might need to add new action hooks here.

Reply:
Added this in the `it_should_update_order_status` test